docs_src/security/Ch-Secure-Consul.md

```diff
@@ -0,0 +1,82 @@
+    default_policy = "deny"
+    enable_token_persistence = true
+}
+```
+
+and this is set as part of EdgeX `security-bootstrapper` service's process.
+The first main step of `security-bootstrapper` for Consul is to bootstrap Consul's ACL system.
+Once Consul's ACL is successfully bootstrapped, `security-bootstrapper` stores the Consul's ACL bootstrap token
+onto the pre-configured folder under `/tmp/edgex/consul/acl-bootstrap-token` or something like that.
+
+`security-bootstrapper` service then will use that token to generate Vault's role based from Consul Secrets Engine API
+`/consul/creds/role/<role-name>` for all internal default EdgeX services and some add-on services via environment variable `ADD_REGISTRY_ACL_ROLES` (see more details and examples in [Configuring Add-on Service documentation section](Ch-Configuring-Add-On-Services.md)).
+
+`security-bootstrapper` then automatically associated with Consul's ACL policy rules with this provided ACL role so that
+Consul token will be created or generated with that ACL rules and hence enforced access controls by Consul
+when the service is communicating with it.
+
+Note that Consul token is generated via Vault's `/consul/creds/` API with Vault's secretstore token and hence
+the generated Consul token is inherited the time-restriction nature from Vault system itself. Thus Consul token
+will be revoked by Vault if Vault's token used to generate it gets expired.  Currently in EdgeX we utilize
+the auto-renewal feature of Vault's token implemented in `go-mod-secrets` to keep Consul token alive and not expired.
+
+## Get Consul ACL token
+
+Consul's access token can be obtained from the `compose-builder` of `edgex-compose` repository via command `make get-consul-acl-token`.  One example of this will be like:
+
+```console
+$ make get-consul-acl-token 
+ef4a0580-d200-32bf-17ba-ba78e3a546e7
+```
+
+This output token is Consul's ACL bootstrap token and thus one can use it to login and access
+Consul service's features from Consul's GUI on http://localhost:8500/ui.
+
+From the upper right-hand corner of Consul's GUI or the "Log in" button in the center,
+one can login with the obtained Consul token in order to access Consul's GUI features:
+
+![Consul-login-GUI](consul-login.png)
+
+![Consul-login-input](consul-login-input.png)
+
+If the end user wants to access consul from the command line and since by default now Consul is running in
+ACL enabled mode, any API call to Consul's endpoints will requires the access token
+and thus one needs to give the access token into the header `X-Consul-Token` of HTTP calls.
+
+One example using `curl` command with Consul access token to do local Consul KV store is given as follows:
+
+```console
+curl -v -H "X-Consul-Token:8775c1db-9340-d07b-ac95-bc6a1fa5fe57" -X PUT --data 'TestKey="My key values"' \
+    http://localhost:8500/v1/kv/my-test-key
+```
+
+where the Consul access token is passed into the header `X-Consul-Token` and assuming it has `written` permission
```

Comment:
"written"  permission?

Reply:
tied to ACL policy. can remove the permission part if it causes confusion.

Reply:
"written" sounds like bad grammar.  Are you sure it isn't "write"?

Reply:
ok, fixed.